src/DnDList.elm

```diff
@@ -711,15 +731,18 @@ stateUpdate operation dropIndex state =
             { state | dragCounter = 0 }
 
 
-listUpdate : Operation -> DragIndex -> DropIndex -> List a -> List a
-listUpdate operation dragIndex dropIndex list =
+listUpdate : Operation -> DragIndex -> DropIndex -> ElementHalf -> List a -> List a
+listUpdate operation dragIndex dropIndex whichHalf list =
     case operation of
         InsertAfter ->
             Internal.Common.Operations.insertAfter dragIndex dropIndex list
 
         InsertBefore ->
             Internal.Common.Operations.insertBefore dragIndex dropIndex list
 
+        InsertAround ->
+            Internal.Common.Operations.insertAround dragIndex dropIndex whichHalf list
+
         Rotate ->
```

Comment:
I would add the new argument at the beginning, so the `dragIndex dropIndex list` triad stay close to each other:
`Internal.Common.Operations.insertAround whichHalf dragIndex dropIndex list`